contracts/mocks/docs/ERC4626Fees.sol

```diff
@@ -8,6 +8,10 @@ import {SafeERC20} from "../../token/ERC20/utils/SafeERC20.sol";
 import {Math} from "../../utils/math/Math.sol";
 
 /// @dev ERC-4626 vault with entry/exit fees expressed in https://en.wikipedia.org/wiki/Basis_point[basis point (bp)].
+///
+/// NOTE: The contract charges fees in terms of assets, not shares. This means that the fees are calculated based on the
+/// amount of assets that are being deposited or withdrawn, and not based on the amount of shares that are being minted or
+/// redeemed. This is an opinionated design decision that should be taken into account when integrating this contract.
```

Comment:
> taken into account when integrating this contract

we should instead say this contract should not be integrated with and that is not ready for production usage nor audited.

Reply:
I would say it's ready for production usage under certain conditions that are up to the developer. Instead, I'd focus on a list of considerations to take before using it, I feel the current note is already precise enough, wdyt?

Reply:
imo: not audited = not production ready, unless the dev wants to take the risk. also we have the considerations that `maxWithdraw` and `maxRedeem` are not considering fees, so the `ERC4626Exceeded` checks don't work correctly.

Reply:
I agree with Monchi, this is in mocks, and its not released in packages. It should be used "at your own risks", similar to the community contracts.

Reply:
Agree with both, I just think it's more precise to prompt people to _use under their own responsibility since it hasn't been audited_ than warning about _not being production ready_

> also we have the considerations that maxWithdraw and maxRedeem are not taking into account fees

Also true. We can clarify that `maxWithdraw` and `maxRedeem` are capped to the user's balance; I don't see a way where it could fail other than if they're customized.